tools/run_third_party_dart.py

```diff
@@ -3,8 +3,12 @@
 # found in the LICENSE file.
 
 import subprocess
+import os
 
 if __name__ == '__main__':
-    subprocess.call(["src/third_party/dart/tools/sdks/dart-sdk/bin/pub", "global", "activate", "-spath", "./src/flutter/tools/generate_package_config"])
-    subprocess.call(["src/third_party/dart/tools/sdks/dart-sdk/bin/pub", "global", "run", "generate_package_config:generate_from_legacy", "src/flutter/flutter_frontend_server/.packages"])
-    subprocess.call(["src/third_party/dart/tools/sdks/dart-sdk/bin/pub", "global", "run", "generate_package_config:generate_from_legacy", "src/flutter/tools/const_finder/.packages"])
+    pub = "src/third_party/dart/tools/sdks/dart-sdk/bin/pub"
+    if os.name == "nt":
+        pub = "src/third_party/dart/tools/sdks/dart-sdk/bin/pub.bat"
```

Comment:
Actually I just tried this on my local Windows box and it only seems to work with backslashes, like:
`subprocess.call(["bin\\cache\\dart-sdk\\bin\\pub.bat"])`

Reply:
It was working for several weeks when I was using `dart pub` on windows

Reply:
switched to backslashes

Reply:
nit: use os.path.join

Reply:
And then you can just conditionally define a string called `BAT` based on whether it's nt and append that to the output of path.join.